### PR TITLE
Sets clippy to pedantic and fixes all clippy warnings 

### PR DESCRIPTION
**Pull request by yara-blue**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,6 +26,7 @@ regex = "1.6.0"
 
 [dev-dependencies]
 float_eq = "0.7"
+built = "0.5"
```
Comment (yara-blue) [inline Cargo.toml:29]: used to detect CI env in a single test

File: src/main.rs [hunk 2]
```diff
@@ -13,19 +16,33 @@ use time::{OffsetDateTime, Time};
 use directory::Uuid;
 use util::AcceptErr;
 
-use crate::util::time::{set_os_timezone, should_lock, try_to_time};
+use crate::util::time::{set_os_timezone, should_lock, try_to};
```
Comment (yara-blue) [inline src/main.rs]: should be used as time::try_to

File: src/report.rs [hunk 3]
```diff
@@ -256,12 +259,13 @@ pub fn remove() -> Result<()> {
...
     use crate::directory::test::test_tree;
-    use crate::directory::Uuid;
 
     #[test]
     pub fn pdf() -> Result<()> {
+        simplelog::SimpleLogger::init(log::LevelFilter::Warn, simplelog::Config::default())
+            .unwrap();
         let tree = test_tree();
-        let roots = vec![*tree.root(Uuid::from(""))];
+        let roots = vec![*tree.root()];
```
Comment (yara-blue) [inline src/report.rs:268]: `root(&self)` previously took a `Uuid` as arg, the idea was that there would be multiple roots (*trash* for example is a root). This is not used and only the root `""` (empty string) is used. Therefore this change.

File: src/systemd.rs [hunk 4]
```diff
@@ -6,22 +6,26 @@ use std::{fs, thread};
 use color_eyre::eyre;
 use eyre::{eyre, Result, WrapErr};
 
-use crate::util::time::try_to_time;
+use crate::util::time::try_to;
```
Comment (yara-blue) [inline src/systemd.rs]: this import should also be limited to time such that usage reads like: `time::try_to`

File: src/systemd.rs [hunk 5]
```diff
@@ -53,16 +55,17 @@ fn is_active(service: &str) -> Result<bool> {
     Ok(output.status.code().unwrap() == 0)
 }
 
-fn wait_for(service: &str, state: bool) -> Result<()> {
+fn wait_for(service: &str, wanted_active: bool) -> Result<()> {
     for _ in 0..20 {
-        if state == is_active(service)? {
```
Comment (yara-blue) [inline src/systemd.rs:58]: a match over a tuple would be a lot more readable here

File: src/systemd.rs [hunk 6]
```diff
@@ -118,8 +121,8 @@ pub fn write_service() -> Result<()> {
 
 // String should be written to a .timer file
 fn timer_str(args: &crate::Args) -> Result<String> {
-    let start = try_to_time(&args.start).wrap_err("Invalid start time")?;
-    let end = try_to_time(&args.end).wrap_err("Invalid end time")?;
+    let start = try_to(&args.start).wrap_err("Invalid start time")?;
+    let end = try_to(&args.end).wrap_err("Invalid end time")?;
```
Comment (yara-blue) [inline src/systemd.rs]: add the `time::` discussed above

File: src/util/time.rs [hunk 7]
```diff
@@ -7,7 +7,7 @@ use rust_fuzzy_search::fuzzy_search_best_n;
 use std::{io::BufRead, process::Command};
 use time::Time;
 
-pub fn try_to_time(s: &str) -> Result<time::Time> {
+pub fn try_to(s: &str) -> Result<time::Time> {
```
Comment (yara-blue) [inline src/util/time.rs]: add `time::`

Comment (yara-blue) [reply src/util/time.rs]: instaed moved to trait

File: src/systemd.rs [hunk 8]
```diff
@@ -53,17 +56,22 @@ fn is_active(service: &str) -> Result<bool> {
...
-        true => Err(eyre!("Time out waiting for activation")),
-        false => Err(eyre!("Time out waiting for deactivation")),
-    }
+
+    Err(eyre!(
+        "Time out waiting for {}",
+        match wanted_active {
+            true => "activation",
+            false => "deactivation",
+        }
```
Comment (evavh) [inline src/systemd.rs:74]: I'm with clippy here, this should be an if-else.

**Review by evavh — CHANGES_REQUESTED** [review]

File: src/systemd.rs (hunk 8 above)
Comment (yara-blue) [reply src/systemd.rs:74]: good catch

**Review by evavh — APPROVED** [review]

**Pull request merged**